src/axmysql/projection/offsetstore.go

```diff
@@ -73,6 +74,30 @@ func (OffsetStore) SaveOffset(
 		return err
 	}
 
+	var no uint64
+	err := tx.QueryRowContext(
+		ctx,
+		`SELECT
+			next_offset
+		FROM ax_projection_offset
+		WHERE persistence_key = ?
+		FOR UPDATE`,
+		pn,
+	).Scan(
+		&no,
+	)
+	if err != nil {
```

Comment:
There needs to be a check here for `sql.ErrNoRows`. When that occurs we want to return the error about the offsets not matching, not the no-rows error itself. 

See https://github.com/jmalloc/ax/blob/fd9770ecf2d3b5cf234c75be0a707f247ecfc593/src/axmysql/saga/crud.go#L171 for an example.